test/unit/fhir_models_test.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+require File.expand_path '../test_helper.rb', __dir__
+
+describe FHIR::Models do
+  describe '#from_contents' do
+    it 'should set source_contents' do
+      bundle_json = File.read('test/fixtures/bundle_1.json')
+      bundle_resource = FHIR.from_contents(bundle_json)
+      bundle_resource.entry.each do |entry|
+        assert entry.source_contents.present?, "entry.source_contents not populated for #{entry}"
+        assert_instance_of String, entry.source_contents
+        assert entry.resource.source_contents.present?, "entry.resource.source_contents not populated for #{entry}"
+        assert_instance_of String, entry.resource.source_contents
+      end
+    end
+
+    it 'should preserve primitive extensions in source_contents' do
+      procedure_json = File.read('test/fixtures/procedure_primitive_extension.json')
```

Comment:
I think we need to test preserving primitive extensions in bundle entries and contained resources.

Reply:
Sure, I can add that.

Reply:
The bundle entry test looks good, but I don't see a test for contained resources https://www.hl7.org/fhir/references.html#contained

Reply:
Updated to contain a test for contained resources as well. Thanks!